packages/aws-cdk-lib/aws-ecr/lib/repository.ts

```diff
@@ -723,7 +732,7 @@ export class Repository extends RepositoryBase {
       resourceName: this.physicalName,
     });
 
-    if (props.autoDeleteImages) {
+    if (props.emptyOnDelete === undefined && props.autoDeleteImages) {
```

Comment:
Given https://github.com/aws/aws-cdk/pull/28233#issuecomment-1844601555.
We should validate for `emptyOnDelete` as well:
```ts
if (props.emptyOnDelete) {
  if (props.removalPolicy !== RemovalPolicy.DESTROY) {
    throw new Error('Cannot use \'emptyOnDelete\' property on a repository without setting removal policy to \'DESTROY\'.');
  }
  this.enableAutoDeleteImages();
} 
if (props.emptyOnDelete === undefined && props.autoDeleteImages) {
  ...
}
```
With a related unit test for coverage.